src/plugins/auto/docs/integration.md

```diff
@@ -1,5 +1,59 @@
 # AUTO Plugin Integration
 
-## How to integrate a new hardware plugin 
+## Implement a new plugin
+Please refer to [OpenVINO Plugin Developer Guide](https://docs.openvino.ai/latest/openvino_docs_ie_plugin_dg_overview.html)
 
-TBD-100527
\ No newline at end of file
+QueryNetwork(query_model) is recommanded since it is important for AUTO to make quick decision and save selection time.
+
+## AUTO Plugin requirements of properties
+
+.. note:: AUTO Plugin call IE APIs to plugins
+
+| IE API                                       | OV API 2.0                           | Mandatory                             |
+| -------------------------------------------- | ------------------------------------ | ------------------------------------- |
+| CONFIG_KEY(DEVICE_ID)                        | ov::device::id                       | Yes                                   |
+| CONFIG_KEY(PERF_COUNT)                       | ov::enable_profiling                 | Yes                                   |
+| CONFIG_KEY(PERFORMANCE_HINT)                 | ov::hint::performance_mode           | Yes                                   |
+| CONFIG_KEY(PERFORMANCE_HINT_NUM_REQUESTS)    | ov::hint::num_requests               | Yes                                   |
+| METRIC_KEY(FULL_DEVICE_NAME)                 | ov::device::full_name                | Yes                                   |
+| METRIC_KEY(NETWORK_NAME)                     | ov::model_name                       | Yes                                   |
+| METRIC_KEY(OPTIMAL_BATCH_SIZE)               | ov::optimal_batch_size               | No: Have this for size decision.      |
+| METRIC_KEY(OPTIMAL_NUMBER_OF_INFER_REQUESTS) | ov::optimal_number_of_infer_requests | Yes                                   |
+| METRIC_KEY(RANGE_FOR_STREAMS)                | ov::range_for_streams                | Yes                                   |
+| METRIC_KEY(SUPPORTED_CONFIG_KEYS)            | ov::supported_properties             | Yes                                   |
+| METRIC_KEY(SUPPORTED_METRICS)                | ov::supported_properties             | Yes                                   |
+| METRIC_KEY(OPTIMIZATION_CAPABILITIES)        | ov::device::capabilities             | Yes                                   |
+| METRIC_KEY(DEVICE_GOPS)                      | ov::device::gops                     | No: For automatic device selection priority. |
+| METRIC_KEY(DEVICE_ARCHITECTURE)              | ov::device::architecture             | No                                    |
```

Comment:
for what reason auto calls device::architecture?
can we write reason for each property? Instead of boolean mandatory flag

BTW, should we convert these requirements for HW plugins into plugin development guide? Eg. create a dedicated page here https://docs.openvino.ai/latest/openvino_docs_ie_plugin_dg_overview.html